### PR TITLE
tests(jiva-operator): add tests for for jiva csi based volumes

### LINKED ISSUE #45: Add e2e pipeline tests for jiva csi and operators
**Describe the problem/challenge you have**
Add e2e tests in jiva csi and operator pipelines

### PR DESCRIPTION
**Pull request by payes**
fixes: https://github.com/openebs/jiva-operator/issues/45
This PR modifies the existing jiva related tests to test jiva csi based volumes also.
Following tests have been updated:
- PBCY-NFSv4-JIVA-PROVISION
- Jiva-Volume-Scaleup
- Jiva-Data-Integrity
- Jiva-App-Scaleup
- Jiva-App-Target-affinity
- Jiva-Memory-Consumption
- Jiva-delete-autogen-snapshot
- Backup-Restore-Jiva
- App-Deploy-Jiva-RWX
- Jiva-single-replica-failure
- Jiva-controller-kill
- Jiva-controller-network-delay
- Jiva-replica-failure
- Jiva-replica-node-affinity
- nfs-App-Kill-Jiva
- nfs-Ctrl-Kill-Jiva
- nfs-Ctrl-Container-Kill-Jiva
- Jiva-controller-failure-replica-scaleup
- Jiva-Revision-Counter
- nfsv4-jiva-provision-cleanup
- Resize
- ResizeXFS

Signed-off-by: Payes Anand <payes.anand@mayadata.io>

### CODE REVIEW

File: experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml [hunk 1]
```diff
@@ -79,6 +88,12 @@
    - name: Getting the master replica pod name corresponding to the IP
      shell: kubectl get pod -l openebs.io/replica=jiva-replica,openebs.io/persistent-volume={{ pv_name }} -n {{ operator_ns }} --no-headers -o jsonpath='{.items[?(@.status.podIP=="{{ replica_ip }}")].metadata.name}'
      register: replica_name
+     when: storage_class != 'jiva-csi-sc'
+
+   - name: Getting the master replica pod name corresponding to the IP
```
Comment (nsathyaseelan) [inline experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml:93]: @payes  this task is duplicate of the previous task.

File: providers/jiva-csi-provisioner/run_litmus_test.yml [hunk 2]
```diff
@@ -9,14 +9,14 @@ spec:
...
-        app: jiva-csi-provisioner
+        provider: jiva-operator
     spec:
       serviceAccountName: litmus
       restartPolicy: Never
       containers:
       - name: ansibletest
         image: openebs/ansible-runner:ci
-        imagePullPolicy: IfNotPresent
+        imagePullPolicy: Always
```
Comment (nsathyaseelan) [inline providers/jiva-csi-provisioner/run_litmus_test.yml]: can you use the Imagepullpolicy is IfNotPresent as default

File: chaoslib/openebs/jiva_csi_controller_network_delay.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    kubectl get pods -l openebs.io/component=jiva-controller
+    -n {{ operator_ns }} --no-headers | grep {{ pv.stdout }} 
+    | awk '{print $1}'
+  args:
+    executable: /bin/bash
+  register: jiva_controller_pod
+
+- name: Record the jiva controller container name
+  set_fact:
+    # Depends on the naming convention in maya-apiserver (<pv-id>-rep)
```
Comment (nsathyaseelan) [inline chaoslib/openebs/jiva_csi_controller_network_delay.yaml]: is this comment is required here?

File: experiments/chaos/nfs_chaos/nfs_openebs_target_failure/test_prerequisites.yml [hunk 4]
```diff
@@ -45,6 +45,20 @@
         stg_engine: "{{ openebs_stg_engine.stdout }}"
   when: stg_prov == "openebs.io/provisioner-iscsi"
 
+- block:
+    - name: Derive PV name from PVC to query storage engine type (openebs)
```
Comment (nsathyaseelan) [inline experiments/chaos/nfs_chaos/nfs_openebs_target_failure/test_prerequisites.yml]: @payes  why this task is needed here

File: experiments/chaos/openebs_single_replica_failure/run_litmus_test.yml [hunk 5]
```diff
@@ -28,5 +28,8 @@ spec:
           - name: APP_LABEL
             value: 'app=cassandra'
 
+          - name: STG_ENGINE
```
Comment (nsathyaseelan) [inline experiments/chaos/openebs_single_replica_failure/run_litmus_test.yml:31]: Is this test is verified with both jiva and jiva csi as it involves the changes in env

File: experiments/chaos/openebs_target_failure/chaosutil.j2 [hunk 6]
```diff
@@ -7,12 +7,10 @@
     {% endif %}
   {% endif %}
 {% endif %}
-{% if stg_prov is defined and (stg_prov == 'openebs.io/provisioner-iscsi' or stg_prov == 'jiva.csi.openebs.io') %}
-  {% if stg_engine is defined and stg_engine == 'jiva' %}
+{% if stg_prov is defined and (stg_prov == 'jiva.csi.openebs.io') %}
```
Comment (nsathyaseelan) [inline experiments/chaos/openebs_target_failure/chaosutil.j2:10]: @payes  In this chaos template the condition to call jiva related utils are removed?

File: experiments/chaos/openebs_target_network_delay/test_prerequisites.yml [hunk 7]
```diff
@@ -67,6 +67,20 @@
       stg_engine: "{{ openebs_stg_engine.stdout }}"
   when: stg_prov == "cstor.csi.openebs.io"
 
+- block:
+  - name: Derive PV name from PVC to query storage engine type (openebs)
```
Comment (nsathyaseelan) [inline experiments/chaos/openebs_target_network_delay/test_prerequisites.yml]: @payes  here also is this task is needed?

File: experiments/chaos/openebs_target_failure/test_prerequisites.yml [hunk 8]
```diff
@@ -86,7 +86,7 @@
 
   - name: Record the storage engine name
     set_fact:
-      stg_engine: "{{ openebs_stg_engine.stdout }}"
+      stg_engine: "jiva-csi"
```
Comment (nsathyaseelan) [inline experiments/chaos/openebs_target_failure/test_prerequisites.yml:73]: @payes if we didn't get the stg_engine name from storage class the above two task is not needed

File: experiments/chaos/openebs_target_network_loss/test.yml [hunk 9]
```diff
@@ -219,7 +218,10 @@
                 containername: "{{ pv.stdout }}-ctrl-con"
               when: 
                 - stg_prov == 'openebs.io/provisioner-iscsi'
+          when:
```
Comment (nsathyaseelan) [inline experiments/chaos/openebs_target_network_loss/test.yml:219]: we can use only one when condition here, like below
when:
    - stg_prov
    -stg_engine

**Review by nsathyaseelan — CHANGES_REQUESTED** [review]

File: providers/jiva-csi-provisioner/test.yml [hunk 10]
```diff
@@ -16,48 +16,37 @@
...
+                dest: "{{ item.dest }}"
+              with_items:
+                  - { src: jiva-csi-sc.j2,dest: jiva-csi-sc.yml }
+                  - { src: jiva-csi-xfs-sc.j2,dest: jiva-csi-xfs-sc.yml }
+                  - { src: local-pv-hostpath.j2,dest: local-pv-hostpath.yml }
+
+            - name: Download the Jiva operator spec
               get_url:
-                url: https://raw.githubusercontent.com/openebs/jiva-csi/master/deploy/jiva-csi-ubuntu-16.04.yaml
+                url: https://raw.githubusercontent.com/payes/jiva-operator/test/deploy/operator.yaml
```
Comment (w3aman) [inline providers/jiva-csi-provisioner/test.yml]: check url path.
use openebs/jiva-operator repo

**Review by w3aman — CHANGES_REQUESTED** [review]

File: experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml (hunk 1 above)
Comment (payes) [reply experiments/functional/snapshot_deletion/jiva/test_replica_restart.yml:93]: This is for CSI-based volumes and the previous task was for non-csi based volumes.
```storage_class != 'jiva-csi-sc'```

File: chaoslib/openebs/jiva_csi_controller_network_delay.yaml (hunk 3 above)
Comment (payes) [reply chaoslib/openebs/jiva_csi_controller_network_delay.yaml]: removed

File: experiments/chaos/nfs_chaos/nfs_openebs_target_failure/test_prerequisites.yml (hunk 4 above)
Comment (payes) [reply experiments/chaos/nfs_chaos/nfs_openebs_target_failure/test_prerequisites.yml]: removed

File: experiments/chaos/openebs_single_replica_failure/run_litmus_test.yml (hunk 5 above)
Comment (payes) [reply experiments/chaos/openebs_single_replica_failure/run_litmus_test.yml:31]: yes, this is verified with both, as this env is newly added and with default value as jiva and not jiva-csi, testing of the old way of provisioning jiva volumes is not affected.

File: experiments/chaos/openebs_target_network_delay/test_prerequisites.yml (hunk 7 above)
Comment (payes) [reply experiments/chaos/openebs_target_network_delay/test_prerequisites.yml]: removed

File: experiments/chaos/openebs_target_failure/test_prerequisites.yml (hunk 8 above)
Comment (payes) [reply experiments/chaos/openebs_target_failure/test_prerequisites.yml:73]: removed

File: experiments/chaos/openebs_target_failure/chaosutil.j2 (hunk 6 above)
Comment (payes) [reply experiments/chaos/openebs_target_failure/chaosutil.j2:10]: added back, thanks.

File: experiments/functional/scale_jiva_csi_replica/README.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
+## Experiment Metadata
+
+| Type       | Description                   | Storage | Applications | K8s Platform |
+| ---------- | ----------------------------- | ------- | ------------ | ------------ |
+| Functional | Scaling jiva storage replicas | OpenEBS | Cassandra    | Any          |
```
Comment (nsathyaseelan) [inline experiments/functional/scale_jiva_csi_replica/README.md]: can we remove the supported Application as Cassandra default?

**Review by nsathyaseelan — CHANGES_REQUESTED** [review]

**Review by nsathyaseelan — APPROVED** [review]
lgtm

**Pull request merged**